File: victorious/victorious/UI/Streams/VStreamCollectionViewController+TrendingShelfResponder.m
//
//  VStreamCollectionViewController+TrendingShelfResponder.m
//  victorious
//
//  Created by Sharif Ahmed on 8/13/15.
//  Copyright (c) 2015 Victorious. All rights reserved.
//

#import "VStreamCollectionViewController+TrendingShelfResponder.h"
#import "victorious-Swift.h"

@implementation VStreamCollectionViewController (TrendingShelfResponder)

- (void)navigateTo:(VStreamItem *__nullable)streamItem fromShelf:(Shelf *__nonnull)fromShelf
{
    [self navigateToStream:fromShelf atStreamItem:streamItem];
}

- (void)trendingUserShelfSelected:(VUser *__nonnull)user fromShelf:(UserShelf *__nonnull)fromShelf
{
    [self.sequenceActionController showProfile:user fromViewController:self];
}

- (void)trendingHashtagShelfSelected:(NSString *__nonnull)hashtag fromShelf:(HashtagShelf *__nonnull)fromShelf
{
    [self showHashtagStreamWithHashtag:hashtag];
}

@end

File: victorious/victorious/UI/Streams/VStreamCollectionViewController+TrendingShelfResponder.swift
//
//  VStreamCollectionViewController+TrendingShelfResponder.swift
//  victorious
//
//  Created by Tian Lan on 9/23/15.
//  Copyright © 2015 Victorious. All rights reserved.
//

extension VStreamCollectionViewController: VShelfStreamItemSelectionResponder, VTrendingUserShelfResponder, VTrendingHashtagShelfResponder {
    func navigateTo(streamItem: VStreamItem?, fromShelf: Shelf) {
        navigateToStream(fromShelf, atStreamItem: streamItem)
    }
    
    func trendingUserShelfSelected(user: VUser, fromShelf: UserShelf) {
        sequenceActionController.showProfile(user)
    }
    
    func trendingHashtagShelfSelected(hashtag: String, fromShelf: HashtagShelf) {
        showHashtagStreamWithHashtag(hashtag)
    }
}
